src/main/java/ca/on/oicr/gsi/dimsum/service/RunListManager.java

```diff
@@ -26,66 +20,15 @@ public class RunListManager {
   private List<Run> runs = new ArrayList<>();
 
   public void update(Map<String, RunAndLibraries> data) {
-    List<Run> newRuns = new ArrayList<>();
-    for (Run run : runs) {
-      String runName = run.getName();
-      RunAndLibraries runAndLibraries = data.get(runName);
-      if (runAndLibraries == null) {
-        log.warn("Orphaned run - run not found: {}", runName);
-        continue;
-      }
-      if (needsQc.test(runAndLibraries)) {
-        log.debug("Maintaining run {}", runName);
-      } else {
-        log.debug("Clearing run {}", runName);
-      }
-    }
-    List<Long> processedRunIds = runs.stream().map(x -> x.getId()).toList();
-    data.values().stream()
-        .filter(needsQc)
-        .filter(x -> !processedRunIds.contains(x.getRun().getId()))
-        .forEach(x -> {
-          log.debug("Creating run {}", x.getRun().getName());
-          newRuns.add(x.getRun());
-        });
-    this.runs = newRuns;
-  }
-
-  public TableData<Run> getRuns(int pageSize, int pageNumber, RunSort sort, boolean descending,
-      RunFilter baseFilter, Collection<RunFilter> filters) {
-    List<Run> baseRuns = runs;
-    Stream<Run> stream = applyFilters(baseRuns, filters);
-    if (sort == null) {
-      sort = RunSort.COMPLETION_DATE;
-      descending = true;
-    }
-    stream = stream.sorted(descending ? sort.comparator().reversed() : sort.comparator());
-    List<Run> filteredRuns =
-        stream.skip(pageSize * (pageNumber - 1)).limit(pageSize).collect(Collectors.toList());
-    TableData<Run> data = new TableData<>();
-    data.setTotalCount(baseRuns.size());
-    data.setFilteredCount(applyFilters(baseRuns, filters).count());
-    data.setItems(filteredRuns);
-    return data;
+    // add new runs to existing list of runs
+    data.values().stream().forEach(x -> {
+      log.debug("Creating run {}", x.getRun().getName());
+      this.runs.add(x.getRun());
+    });
   }
 
-  private Stream<Run> applyFilters(List<Run> runs, Collection<RunFilter> filters) {
-    Stream<Run> stream = runs.stream();
-    if (filters != null && !filters.isEmpty()) {
-      Map<RunFilterKey, Predicate<Run>> filterMap = new HashMap<>();
-      for (RunFilter filter : filters) {
-        RunFilterKey key = filter.getKey();
-        if (filterMap.containsKey(key)) {
-          filterMap.put(key, filterMap.get(key).or(filter.predicate()));
-        } else {
-          filterMap.put(key, filter.predicate());
-        }
-      }
-      for (Predicate<Run> predicate : filterMap.values()) {
-        stream = stream.filter(predicate);
-      }
-    }
-    return stream;
+  public List<Run> getRuns() {
+    return this.runs;
```

Comment:
There's no need to maintain this separate list of runs - can get rid of this class entirely. `CaseService` can get runs from the case data: `caseData.getRunsAndLibraries().stream().map(RunAndLibraries::getRun)`